drift.py

```diff
@@ -1,140 +1,77 @@
-                # Save concrete spec hash as last hash.
-                self.last[spec] = concrete_spec
-            else:
-                # If the spec doesn't concretize properly we also want
-                # to record the commit at which this occurred.
-                result.tags += ["concretizaton-failed"]
-                send(result)
-
-    def tag_deps(self, concrete_spec, spec, name):
-        tags = []
-        # Build dependency cache map for the last spec.
-        cache_map = {}
-        for i in range(len(self.last[spec])):
-            for name, dep in self.last[spec][i].items():
-                cache_map[name] = dep
-        # Iterate through all of the dependencies in the new spec.
-        for i in range(len(concrete_spec)):
-            for name, dep in concrete_spec[i].items():
-                # Check if a dependency has been added.
-                if name not in cache_map:
-                    tags += ["dep-added"]
-                    continue
-                # Check if a dependency version is different.
-                elif dep["full_hash"] != cache_map[name]["full_hash"] and name != spec:
-                    tags += ["dep-updated"]
-                # Remove dependency from known dependencies to track if we don't hit any
-                # dependencies in the new version that use to be in the old spec.
-                cache_map.pop(name)
-        # If there are dependencies that we didn't hit in the new version mark
-        # the commit has having lost dependencies.
-        if len(cache_map) > 0:
-            tags += ["dep-deleted"]
-        return tags
-
-    def tag_variants(self, concrete_spec, spec, name):
-        tags = []
-        if "parameters" in concrete_spec[0][name]:
-            # Check for variants added or modified by comparing new spec to old.
-            for param, value in concrete_spec[0][name]["parameters"].items():
-                if param not in self.last[spec][0][name]["parameters"]:
-                    if param == "patches":
-                        tags += ["patches-added"]
-                    else:
-                        tags += ["variant-added"]
-                elif value != self.last[spec][0][name]["parameters"][param]:
-                    if param == "patches":
-                        tags += ["patches-modified"]
-                    else:
-                        tags += ["variant-modified"]
-            # Compare old to new to see if any variants were removed.
-            for param, value in self.last[spec][0][name]["parameters"].items():
-                if param not in concrete_spec[0][name]["parameters"]:
-                    if param == "patches":
-                        tags += ["patches-removed"]
-                    else:
-                        tags += ["variant-removed"]
-            return tags
+        for abstract_spec in self.specs:
+            # Separate out spec version and name.
+            verToken = abstract_spec.find("@")
```

Comment:
Can you walk me through what you are trying to do here? Do you just need to get a name and version from the string? This looks very Go oriented, and there are much easier ways to do this in Python! But actually spack can help us to make this much easier:

```python
import spack.Spec
spec = spack.Spec(abstract_spec)
```
And then you can literally run the same code below, instead of trying to capture output, just through spack directly, and manipulate the spec object as needed.

Reply:
If we run `spec = space.Spec(abstract_spec) won't that cause Spack to load the `package.py` file here every time? I worry that we'll end up DDOS-ing our own network file system? But yes you're exactly right I'm just looking to split up the abstract spec into the name and version strings.

Reply:
If spack has already loaded packages I don't think so? if you really just want to split, just do:

```python
name = abstract_spec
version = None
if "@" in abstract_spec:
    name, version = abstract_spec.split("@", 1)
``

Reply:
Oh gotcha! That is much cleaner!